src/components/SingleVideo/SingleVideo.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+import { usePlaylist } from "../../contexts";
+import {
+  addToWatchLater,
+  removeFromWatchLater,
+  isVideoInWatchLater,
+} from "../../utils";
+import { Modal } from "../";
+import styles from "./SingleVideo.module.css";
+
+export const SingleVideo = () => {
+  const [video, setVideo] = useState({});
+  const [loader, setLoader] = useState(false);
+  const { videoId } = useParams();
+  const { showModal, openModal, watchLater, playlistDispatch } = usePlaylist();
+  const videoInWatchLater = isVideoInWatchLater(videoId, watchLater);
+  const { _id, alt, views, duration, title, avatar, creatorName, description } =
+    video;
+
+  useEffect(() => {
+    (async () => {
+      try {
+        setLoader(true);
+        const {
+          data: { video },
+        } = await axios.get(`/api/video/${videoId}`);
+
+        setVideo(video);
+        setLoader(false);
+      } catch (error) {
+        console.log(error);

```

Comment:
You can display the error on the UI instead.

Reply:
Yes, I will do that